java/client/src/main/java/glide/utils/ArrayTransformUtils.java

```diff
@@ -9,14 +9,28 @@
 public class ArrayTransformUtils {
 
     /**
-     * Converts a map to an array of strings with alternating keys and values.
+     * Converts a map of string keys and values of any type in to an array of strings with alternating
+     * keys and values.
      *
-     * @param args Map of string pairs to convert.
-     * @return Array of strings [key1, value1, key2, value2, ...].
+     * @param args Map of string keys to values of any type to convert.
+     * @return Array of strings [key1, value1.toString(), key2, value2.toString(), ...].
      */
-    public static String[] convertMapToArgArray(Map<String, String> args) {
+    public static String[] convertMapToKeyValueStringArray(Map<String, ?> args) {
         return args.entrySet().stream()
-                .flatMap(entry -> Stream.of(entry.getKey(), entry.getValue()))
+                .flatMap(entry -> Stream.of(entry.getKey(), entry.getValue().toString()))
+                .toArray(String[]::new);
+    }
+
+    /**
+     * Converts a map of string keys and values of any type into an array of strings with alternating
+     * values and keys.
+     *
+     * @param args Map of string keys to values of any type to convert.
+     * @return Array of strings [value1.toString(), key1, value2.toString(), key2, ...].
+     */
+    public static String[] convertMapToValueKeyStringArray(Map<String, ?> args) {
+        return args.entrySet().stream()
+                .flatMap(entry -> Stream.of(entry.getValue().toString(), entry.getKey()))
```

Comment:
You can introduce a second argument to swap fields or nor like
```suggestion
    public static String[] convertMapToStringArray(Map<String, ?> args, boolean swap) {
        return args.entrySet().stream()
                .flatMap(entry -> swap ? Stream.of(entry.getValue().toString(), entry.getKey()
                : Stream.of(entry.getKey(), entry.getValue().toString())))
```

Reply:
I'm actually not a fan of adding boolean flags for configuring like this usually. It's adding another branch instruction and it doesn't really save much in terms of code reuse.